blocks/library/cover-image/index.js

```diff
@@ -62,17 +63,18 @@ registerBlockType( 'core/cover-image', {
 	},
 
 	edit( { attributes, setAttributes, focus, setFocus, className } ) {
-		const { url, title, align, id, hasParallax, hasBackgroundDim } = attributes;
+		const { url, title, align, id, hasParallax, dimRatio } = attributes;
 		const updateAlignment = ( nextAlign ) => setAttributes( { align: nextAlign } );
 		const onSelectImage = ( media ) => setAttributes( { url: media.url, id: media.id } );
 		const toggleParallax = () => setAttributes( { hasParallax: ! hasParallax } );
-		const toggleBackgroundDim = () => setAttributes( { hasBackgroundDim: ! hasBackgroundDim } );
+		const setDimRatio = ( ratio ) => setAttributes( { dimRatio: ratio } );
 		const style = url
 			? { backgroundImage: `url(${ url })` }
 			: undefined;
 		const classes = classnames( className, {
 			'has-parallax': hasParallax,
-			'has-background-dim': hasBackgroundDim,
+			'has-background-dim': dimRatio !== 0,
+			[ dimRatioToClass( dimRatio ) ]: dimRatio !== 0,
```

Comment:
Maybe `dimRatioToClass` should return an empty string or null value when it is not equal to zero, which we pass as a standalone argument to `classnames` to be dropped?

Reply:
Saner, yes.